In [1]:
import pandas as pd
import numpy as np
import warnings

warnings.filterwarnings('ignore')
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows',None)

**Índice**

1. [Limpieza 'inventory'](#id1)
2. [Limpieza 'language'](#id2)
3. [Limpieza 'old_HDD'](#id3)

<div id='id1' />

# Limpieza 'inventory'

In [2]:
df = pd.read_csv('../raw_data/inventory.csv')
df.head()

inventory_id  film_id  store_id          last_update
0             1        1         1  2006-02-15 05:09:17
1             2        1         1  2006-02-15 05:09:17
2             3        1         1  2006-02-15 05:09:17
3             4        1         1  2006-02-15 05:09:17
4             5        1         2  2006-02-15 05:09:17

In [3]:
print(len(df.inventory_id.unique()))
print(len(df.film_id.unique()))
print(len(df.store_id.unique()))
print(len(df.last_update.unique()))

1000
207
2
1


In [4]:
df = df.drop('last_update', axis=1)

De la exploración del inventario encontramos que existe más de una tienda y que en stock existen 1000 películas que corresponden a 207 títulos diferentes.
De nuevo decidimos eliminar la columna 'last_update' debido a que únicamente contiene un valor.

In [5]:
df.to_csv('../cleaned_data/4_clean_inventory.csv' , index=False)

<div id='id2' />

# Limpieza 'language'

In [6]:
df2 = pd.read_csv('../raw_data/language.csv')
df2.head()

language_id      name          last_update
0            1   English  2006-02-15 05:02:19
1            2   Italian  2006-02-15 05:02:19
2            3  Japanese  2006-02-15 05:02:19
3            4  Mandarin  2006-02-15 05:02:19
4            5    French  2006-02-15 05:02:19

In [7]:
df2.describe().T

count  mean       std  min   25%  50%   75%  max
language_id    6.0   3.5  1.870829  1.0  2.25  3.5  4.75  6.0

En la columna 'language' observamos que únicamente se presentan seis valores diferentes para seis idiomas diferentes. No hay valores nulos. De nuevo decicimos dropear la columna 'last_upadte' y exportamos este data frame cómo csv

In [8]:
df2.to_csv('../cleaned_data/5_clean_language.csv' , index=False)

<div id='id3' />

# Limpieza 'old_HDD'

In [9]:
df3 = pd.read_csv('../raw_data/old_HDD.csv')
df3.head()

first_name last_name                  title  release_year  category_id
0   PENELOPE   GUINESS       ACADEMY DINOSAUR          2006            6
1   PENELOPE   GUINESS   ANACONDA CONFESSIONS          2006            2
2   PENELOPE   GUINESS            ANGELS LIFE          2006           13
3   PENELOPE   GUINESS  BULWORTH COMMANDMENTS          2006           10
4   PENELOPE   GUINESS          CHEAPER CLYDE          2006           14

In [10]:
df3.first_name = df3.first_name.apply(lambda x : x.lower())
df3.first_name = df3.first_name.apply(lambda x : x.capitalize())

In [11]:
df3.last_name = df3.last_name.apply(lambda x : x.lower())
df3.last_name = df3.last_name.apply(lambda x : x.capitalize())

In [12]:
df3.title = df3.title.apply(lambda x : x.lower())
df3.title = df3.title.apply(lambda x : x.capitalize())

Del data frame obtenido por old_HDD.csv decidimos aplicar funciones para mejorar la legibilidad de las columnas y observamos que estos datos parecen ser una segunda recopilación de datos antiguos sobre actores, películas, el año y la categoria del film.

In [14]:
df3.to_csv('../cleaned_data/6_clean_old_HDD.csv' , index=False)